app/controllers/concerns/foreman/controller/registration_command.rb

```diff
@@ -0,0 +1,66 @@
+module Foreman::Controller::RegistrationCommand
+  extend ActiveSupport::Concern
+
+  private
+
+  def command
+    args_query = "?#{registration_args.to_query}"
+    "curl #{insecure} -X GET '#{endpoint}#{args_query if args_query != '?'}' #{command_headers} | bash"
+  end
+
+  def registration_args
+    args = registration_params.except(*ignored_query_args)
+
+    args['setup_insights'] = setup_insights.to_s
+    args['setup_remote_execution'] = setup_remote_execution.to_s
+
+    args.delete_if { |_, v| v.blank? }
+        .permit!
+  end
```

Comment:
I don't think this is needed... the same can be achieved within `registration_params` method if all you need is to remove empty/falsy params. Then you can use e.g. `registration_params.to_query`.

BTW, `[false, nil, "", "    ", [], {}].map(&:blank?) #=> [true, true. true, true, true. true]`.

UPD: Sorry, I see now that `setup_insights` return value is used... but still I question why that method exists...

Reply:
> but still I question why that method exists...

Well I didn't want to have too much logic in the `command` method, so I separated it to the `registration_args`.